### ISSUE TITLE
Add Dart 3.3

### ISSUE THREAD

**Issue by Faridalim** [codewars/runner#249]
Please complete the following information:

- Name: Dart
- Version: 3.0.2
- Release Note/Changelog: https://dart.dev/guides/language/evolution#dart-30

---

:+1: reaction _might_ help to get this request prioritized.

**Comment by monadius** [comment]
The first thing which must be done is to update all existing katas to Dart 2.14. Here is a list of katas which must be updated: https://github.com/codewars/content-issues/wiki/List-of-Dart-Kata-to-Update.

**Comment by Faridalim** [comment]
@monadius what can we do to update all katas? Can we directly update them to 3.0.2 and skip 2.14?

**Comment by monadius** [comment]
A general Codewars rule is to keep at most 2 language version at the same time. So a new language version cannot be added until Dart 2.8 is removed. I don't see any issues with updating katas to Dart 2.14 first. Dart 2.14 katas should be mostly compatible with Dart 3.0.2. Ideally, all updates should be done in such a way that katas work with both Dart 2.14 and the newest version of Dart.

**Comment by Faridalim** [comment]
I see. What can i do to contribute? I dont know if my skill level is enough since i am new to Dart hence the use of codewars

**Comment by monadius** [comment]
If you would like to update some Dart 2.8 katas, please follow these steps:

- Open the Details page of a kata which you want to update.
- Click the "Fork" link on the right (this link is available for solved katas only).
- Switch the language version to Dart 2.14 (in the upper right corner) and click the Run button. Most likely, you get some error message. Fix all errors and save changes. 
- Make sure that both "Test Cases" and "Sample Test Cases" work without issues with Dart 2.14. Also verify that the Solution Setup is correct.
- Publish your fork and post a message with the link to your fork in the Discourse page of the kata (something like "Dart 2.14 update: link").
- After that, someone needs to review and approve your changes.

In general, the main problem with existing Dart 2.8 katas is that they are not null safe. Sometimes, explicit argument types are missing for some functions and it is enough to provide them to update a kata to Dart 2.14.

One example: 
```dart
  int rand(min, max) => inst.nextInt(max - min + 1) + min;
```
This function should be changed to
```dart
  int rand(int min, int max) => inst.nextInt(max - min + 1) + min;
```

**Comment by ram6ler** [comment]
Dart is currently at version 3.2.6. Dart 3 has a lot of features not available in previous versions, so it's a shame Dart 3 is still not available on CW...

**Comment by kazk** [comment]
Added Dart 3.3.

All published kata were upgraded to use 3.3 by default. Please let me know if you find any issues.
